backend/inspirehep/matcher/cli.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more detail
+
+import click
+import structlog
+from celery import group
+from flask.cli import with_appcontext
+from invenio_pidstore.models import PersistentIdentifier, PIDStatus
+
+from .tasks import match_references_by_uuids
+
+LOGGER = structlog.getLogger()
+
+
+@click.group()
+def match():
+    """Command group for matcher."""
+
+
+@match.command(help="Match references of all literature records")
+@click.option(
+    "-bs",
+    "--batch-size",
+    default=100,
+    help="The number of records per batch that will be indexed by workers.",
+    show_default=True,
+)
+@click.option(
+    "-dbs",
+    "--db-batch-size",
+    default=1000,
+    help="The size of the chunk of records loaded from the DB, aka yield_per argument",
+    show_default=True,
+)
+@with_appcontext
+def references(batch_size, db_batch_size):
+    # TODO: logs and maybe progress?
+    literature_uuids_query = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_type == "lit",
+        PersistentIdentifier.status == PIDStatus.REGISTERED,
+    ).with_entities(PersistentIdentifier.object_uuid)
+    uuid_chunk = []
+    matcher_tasks = []
+    for result in literature_uuids_query.yield_per(db_batch_size):
+        uuid = result[0]
+        uuid_chunk.append(uuid)
+        if len(uuid_chunk) == batch_size:
```

Comment:
I think this has a bug if the number of results is not an integer multiple of `batch_size`. You can use https://github.com/inspirehep/inspirehep/blob/cea05cab04a1f3d94bee7097a40402c707cebd6c/backend/inspirehep/utils.py#L23

Reply:
oh yeah you're right about the bug, I don't like `chunker` but I'll have a look if it works in this case.